nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/cosmos/document/AbstractCosmosDocumentProcessor.java

```diff
@@ -0,0 +1,329 @@
+
+    static final Relationship REL_ORIGINAL = new Relationship.Builder()
+            .name("original")
+            .description("All input FlowFiles that are part of a successful query execution go here.")
+            .build();
+
+    static final PropertyDescriptor CONNECTION_SERVICE = new PropertyDescriptor.Builder()
+        .name("Cosmos-Document-DB-Connection-Service")
+        .displayName("Cosmos Document DB Connection Service")
+        .description("If configured, this property will use the assigned for retrieving connection string info.")
+        .required(false)
+        .identifiesControllerService(AzureCosmosDocumentConnectionService.class)
+        .build();
+
+
+    static final PropertyDescriptor URI = new PropertyDescriptor.Builder()
+        .name("Cosmos Document DB URI")
+        .displayName("Cosmos Document DB URI")
+        .description("CosmosURI, typically of the form: https://{databaseaccount}.documents.azure.com:443/. Note this host URL is for Cosmos DB with SQL API")
+        .required(false)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .build();
+
+    static final PropertyDescriptor DB_ACCESS_KEY = new PropertyDescriptor.Builder()
+        .name("Cosmos Document DB Access Key")
+        .displayName("Cosmos Document DB Access Key")
+        .description("Cosmos DB Access Key from Azure Portal (Settings->Keys)")
+        .required(false)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .sensitive(true)
+        .build();
+
+    static final PropertyDescriptor DATABASE_NAME = new PropertyDescriptor.Builder()
+        .name("Cosmos Database Name")
+        .displayName("Cosmos Database Name")
+        .description("The name of the database to use")
+        .required(true)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .build();
+
+    static final PropertyDescriptor CONTAINER_ID = new PropertyDescriptor.Builder()
+        .name("Cosmos Container ID")
+        .description("Cosmos Container Id to use")
+        .required(true)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .build();
+
+    static final PropertyDescriptor CONSISTENCY = new PropertyDescriptor.Builder()
+        .name("Cosmos Consistency Level")
+        .displayName("Cosmos Consistency Level")
+        .description("Cosmos Consistency Level to use")
+        .required(false)
+        .allowableValues(CONSISTENCY_STRONG, CONSISTENCY_BOUNDED_STALENESS, CONSISTENCY_SESSION,
+                CONSISTENCY_CONSISTENT_PREFIX, CONSISTENCY_EVENTUAL)
+        .defaultValue(CONSISTENCY_SESSION)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .build();
+
+    static final PropertyDescriptor PARTITION_KEY = new PropertyDescriptor.Builder()
+        .name("Cosmos Partition Key Field Name")
+        .description("Partition Key Field Name defined during Cosmos Collection/Container Creation Time")
```

Comment:
Minor change: Please update all places where you refer it to Cosmos to be Cosmos DB

Reply:
Done in AbstractAzureCosmosDBProcessor and AzureCosmosDBConnectionCtonrollerService. Note the changes of java files.